py/objint_mpz.c

```diff
@@ -411,6 +411,14 @@ mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
     }
 }
 
+mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
+    return _mp_obj_int_get_checked(self_in, mpz_as_int_checked);
+}
+
+mp_uint_t mp_obj_int_get_uint_checked(mp_const_obj_t self_in) {
+    return (mp_uint_t)_mp_obj_int_get_checked(self_in, (bool(*)(const mpz_t *, mp_int_t *))mpz_as_uint_checked);
```

Comment:
Extra cast `(mp_uint_t)` shouldn't be needed since AFAIK C99 gurantees implicit conversions between arithmetic types. However, the casting of the function possibly needs some treatment. It's probably well-defined for all ports in use (didn't check though), but for the sake of never causing nasty problems I'd add an explicit compile-time check that the cast is correct. I think just checking that `sizeof(mp_int_t) == sizeof(mp_uint_t)` is sufficient.

Reply:
Mhm, I'll add some compile-time test for this.

About the cast - IIRC I added it mainly for documentation. If I were to see this code w/o a cast, at a first glance I'd think "wow, this can implicitly sign underflow, that's bad". The cast clarifies the developer's intention.

Reply:
Fair enough but you didn't use the cast in any of the other places where it happens so i wasn't sure what was going in :)

Reply:
Yeah technically `MP_OBJ_SMALL_INT_VALUE` is `mp_int_t` as well. But here the need for a cast was more evident, I think.

Anyway, resolving this, it's irrelevant now with the new code structure.